### PR TITLE
Add cache size stats to file_system_http_cache

### PR DESCRIPTION
**Pull request by ravenblackx**
Commit Message: Add cache size stats to file_system_http_cache
Additional Description: This is a step towards supporting eviction in the cache implementation, with the bonus that it also makes stats available for monitoring.
Risk Level: None, this filter is not in use.
Testing: 100% coverage.
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by ravenblackx** [comment]
/assign-from @envoyproxy/first-pass-reviewers

(I don't know if I can do that!)

File: source/extensions/http/cache/file_system_http_cache/stats.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+namespace FileSystemHttpCache {
+
+/**
+ * All cache stats. @see stats_macros.h
+ *
+ * Note that size_bytes and size_count may drift away from true values, due to:
+ * - Changes to the filesystem may be made outside of the process, which will not be
+ *   accounted for.
+ * - Files completed while pre-cache-purge measurement is in progress may not be counted.
+ * - Changes in file size due to header updates are assumed to be negligible, and are ignored.
```
Comment (mattklein123) [inline source/extensions/http/cache/file_system_http_cache/stats.h:18]: Ok here is my answer. I guess just cross/reference comments if possible.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with comment comments, thanks.

/wait

**Review by mattklein123 — APPROVED** [review]
Cool, thanks.

**Pull request merged**